### PR TITLE
 Chapter 6 MODELING USERS

### PR DESCRIPTION
**Pull request by tranChiVi1412**
• Migrations allow us to modify our application’s data model.
• Active Record comes with a large number of methods for creating and
manipulating data models.
• Active Record validations allow us to place constraints on the data in our
models.
• Common validations include presence, length, and format.
• Regular expressions are cryptic but powerful.
• Defining a database index improves lookup efficiency while allowing enforcement of uniqueness at the database level.
• We can add a secure password to a model using the built-in has_secure_password method.

### CODE REVIEW

**Comment by tranChiVi1412** [comment]
Sử dụng migration và generate để tạo model
mapping datatype bằng Active Record
Các ràng buộc các model trong rails
Regular expressions và has_secure_password

**Comment by tranChiVi1412** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  before_save{email.downcase!}
```
Comment (tham-1781) [inline app/models/user.rb:2]: tại sao lại `email.downcase!` vậy chú

File: db/migrate/20210113092619_add_index_to_users_email.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class AddIndexToUsersEmail < ActiveRecord::Migration[6.0]
```
Comment (tham-1781) [inline db/migrate/20210113092619_add_index_to_users_email.rb:1]: file này ko làm gì hết này

**Pull request merged**